SocialCareCaseViewerApi/V1/Gateways/MongoGateway.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+#nullable enable
+namespace SocialCareCaseViewerApi.V1.Gateways
+{
+    public class MongoGateway : IMongoGateway
+    {
+        private readonly IMongoDatabase _mongoDatabase;
+
+        public MongoGateway()
+        {
+            string mongoConnectionString = Environment.GetEnvironmentVariable("MONGO_CONN_STRING") ??
+                                           @"mongodb://localhost:1433/";
+
+            string databaseName = Environment.GetEnvironmentVariable("SCCV_MONGO_DB_NAME") ?? "social_care_db_name";
+
+            var mongoClient = new MongoClient(new MongoUrl(mongoConnectionString));
+            _mongoDatabase = mongoClient.GetDatabase(databaseName);
+        }
+
+        public void DropCollection(string collectionName)
+        {
+            _mongoDatabase.DropCollection(collectionName);
+        }
+
+        public void InsertRecord<T>(string collectionName, T objToAdd)
+        {
+            var collection = _mongoDatabase.GetCollection<T>(collectionName);
+            collection.InsertOne(objToAdd);
+        }
+
+        public void UpsertRecord<T>(string collectionName, Guid id, T record)
+        {
+            var collection = _mongoDatabase.GetCollection<T>(collectionName);
+            var filter = new BsonDocument("_id", new BsonBinaryData(id, GuidRepresentation.Standard));
+            collection.ReplaceOne(filter, record, new ReplaceOptions { IsUpsert = true });
+        }
+
+        public void DeleteRecordById<T>(string collectionName, Guid id)
+        {
+            var collection = _mongoDatabase.GetCollection<T>(collectionName);
+            var filter = Builders<T>.Filter.Eq("_id", id);
+            collection.DeleteOne(filter);
+        }
+
+        public List<T> LoadRecords<T>(string collectionName)
+        {
+            var collection = _mongoDatabase.GetCollection<T>(collectionName);
+            return collection.Find(new BsonDocument()).ToList();
+        }
+
+        public T LoadRecordById<T>(string collectionName, Guid id)
+        {
+            var collection = _mongoDatabase.GetCollection<T>(collectionName);
+            var filter = Builders<T>.Filter.Eq("Id", id);
```

Comment:
What's the difference between `"Id"` and `"_id"`? 🤔 In `DeleteRecordById`, we filter by `_id` instead.

Reply:
I'm honestly surprised it was working, the Mongo ORM must be taking the field "Id" and converting it to "_id". 
I'll update it to use "_id" though.

The way Mongo works with C# is that you give a property on a class the [BsonId] directive, this tells Mongo it is the Id for the object and maps it to the "_id" field in mongo.

I should later make a little change to the ReadMe for debugging Mongo, you can point MongoDb Compass to our docker mongo instance and see the objects being created and removed in our tests. I'm sure at some point it will help someone to debug something if they can see what is actually being added to Mongo. (this is really just a reminded for me :p )

Reply:
That sounds useful especially because I have no idea what MongoDB Compass is! 😂